src/core/src/service_interfaces/LifecycleManagerARC.py

```diff
@@ -0,0 +1,181 @@
+            return ""
+        lines = out.split("\n")
+        dir_name = lines[0].lstrip()
+        core_state_file_path = self.arc_root + dir_name + self.config_folder_path + self.arc_core_state_file_name
+        return core_state_file_path
+
+    def read_arc_core_sequence(self):
+        self.composite_logger.log_debug("Reading arc extension core sequence...")
+        core_state_file_path = self.get_arc_core_state_file()
+        if not os.path.exists(core_state_file_path) or not os.path.isfile(core_state_file_path):
+            ''' Dummy core sequence in case of arc core state is not found '''
+            completed = True
+            core_sequence = {'number': 0,
+                         'action': self.execution_config.operation,
+                         'completed': str(completed),
+                         'lastHeartbeat': str(self.env_layer.datetime.timestamp()),
+                         'processIds': []}
+            return core_sequence
+        
+        # Read (with retries for only IO Errors) - TODO: Refactor common code
+        for i in range(0, Constants.MAX_FILE_OPERATION_RETRY_COUNT):
+            try:
+                with self.env_layer.file_system.open(core_state_file_path, mode="r") as file_handle:
+                    core_sequence = json.load(file_handle)['coreSequence']
+                    return core_sequence
+            except Exception as error:
+                if i < Constants.MAX_FILE_OPERATION_RETRY_COUNT:
+                    self.composite_logger.log_warning("Exception on arc core sequence read. [Exception={0}] [RetryCount={1}]".format(repr(error), str(i)))
+                    time.sleep(i + 1)
+                else:
+                    self.composite_logger.log_error("Unable to read arc core state file (retries exhausted). [Exception={0}]".format(repr(error)))
+                    raise
+    
+    def lifecycle_status_check(self):
+        self.composite_logger.log_debug("Performing lifecycle status check...")
+        extension_sequence = self.read_extension_sequence()
+        arc_core_sequence = self.read_arc_core_sequence()
+
+        if int(extension_sequence['number']) == int(self.execution_config.sequence_number):
+            self.composite_logger.log_debug("Extension sequence number verified to have not changed: {0}".format(str(extension_sequence['number'])))
+        else:
+            self.composite_logger.log_error("Extension goal state has changed. Terminating current sequence: {0}".format(self.execution_config.sequence_number))
+            self.update_core_sequence(completed=True)   # forced-to-complete scenario | extension wrapper will be watching for this event
+            self.env_layer.exit(0)
+        self.composite_logger.log_debug("Completed lifecycle status check.")
+
+        if arc_core_sequence['completed'] == "False":
+            self.composite_logger.log_warning("ARC agent is currently running, sequence number {0}.".format(str(arc_core_sequence['number'])))
```

Comment:
Same review about updating this comment to reflect that older sequence is still running and we are terminating it.